core/src/main/java/com/sap/ai/sdk/core/DeploymentCache.java

```diff
@@ -107,25 +89,26 @@ private Optional<String> getDeploymentIdByModel(@Nonnull final String modelName)
    * @throws NoSuchElementException if no running deployment is found for the scenario.
    */
   @Nonnull
-  public String getDeploymentIdByScenario(
+  String getDeploymentIdByScenario(
       @Nonnull final ApiClient client,
       @Nonnull final String resourceGroup,
       @Nonnull final String scenarioId)
       throws NoSuchElementException {
-    return getDeploymentIdByScenario(scenarioId)
+    return getDeploymentIdByScenario(resourceGroup, scenarioId)
         .orElseGet(
             () -> {
               resetCache(client, resourceGroup);
-              return getDeploymentIdByScenario(scenarioId)
+              return getDeploymentIdByScenario(resourceGroup, scenarioId)
                   .orElseThrow(
                       () ->
                           new NoSuchElementException(
                               "No running deployment found for scenario: " + scenarioId));
             });
   }
 
-  private Optional<String> getDeploymentIdByScenario(@Nonnull final String scenarioId) {
-    return CACHE.stream()
+  private Optional<String> getDeploymentIdByScenario(
```

Comment:
(minor) It would be nice to stick to either VAVRs `Option<>` or Javas `Optional`. Mixing the two is a bit awkward, and mixing them in public API might be confusing for users. In Cloud SDK we used VAVR because it's more powerful and integrated with `Try` etc.

Reply:
This is not public API it's just a private helper function to reduce code duplication.